libsabledb/src/commands/generic_commands.rs

```diff
@@ -172,6 +176,123 @@ impl GenericCommands {
         };
         Ok(Some(md.value_type()))
     }
+
+    async fn expire(
+        client_state: Rc<ClientState>,
+        command: Rc<RedisCommand>,
+        response_buffer: &mut BytesMut,
+    ) -> Result<(), SableError> {
+        // at least 3 arguments
+        check_args_count!(command, 3, response_buffer);
+        let builder = RespBuilderV2::default();
+
+        // EXPIRE key seconds [NX | XX | GT | LT]
+        let mut iter = command.args_vec().iter();
+
+        // skip "expire"
+        iter.next();
+        let Some(key) = iter.next() else {
+            builder.error_string(
+                response_buffer,
+                "ERR wrong number of arguments for 'expire' command",
+            );
+            return Ok(());
+        };
+
+        let db_id = client_state.database_id();
+        let _unused = LockManager::lock_user_key_exclusive(key, db_id);
+        let generic_db = GenericDb::with_storage(&client_state.database(), db_id);
+
+        // check the remaining arguments
+        match (iter.next(), iter.next()) {
+            (Some(seconds), Some(arg)) => {
+                let arg_lowercase = BytesMutUtils::to_string(arg).to_lowercase();
+
+                let mut expiration = match generic_db.get_expiration(key)? {
```

Comment:
You can change this here to:

```rust
let (mut expiration, has_expiration) = match generic_db.get_expiration(key)? {
  Some(expiration) => {
    (expiration, true)
  }
  None => {
    (Expiration::default(), false)
  };
```

And you can use this later in the `xx` and `nx` cases